src/hipscat_import/file_readers.py

```diff
@@ -0,0 +1,104 @@
+"""File reading generators for common file types."""
+
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from astropy.table import Table
+
+# pylint: disable=too-few-public-methods,too-many-arguments
+
+
+def get_file_reader(
+    file_format,
+    chunksize=500_000,
+    header="infer",
+    schema_file=None,
+    column_names=None,
+    type_map=None,
+    separator=",",
+):
+    """Get a generator file reader for common file types"""
+    if "csv" in file_format:
+        return CsvReader(
+            chunksize=chunksize,
+            header=header,
+            schema_file=schema_file,
+            column_names=column_names,
+            type_map=type_map,
+            separator=separator,
+        ).read
+    if file_format == "fits":
+        return fits_reader
+    if file_format == "parquet":
+        return ParquetReader(chunksize=chunksize).read
```

Comment:
I was thinking the argument could take the object rather than just the method of the object. Could be useful in case we need any extra functionality from the file readers in the future but this works

Reply:
Sounds like a good idea, but also sounds like it's safe to wait until we see more need.